src/GafferOSL/OSLObject.cpp

```diff
@@ -310,9 +316,23 @@ void OSLObject::hashProcessedObject( const ScenePath &path, const Gaffer::Contex
 	{
 		h.append( inPlug()->fullTransformHash( path ) );
 	}
-	if( useAttributesPlug()->getValue() )
+
+	if( gafferAttributes )
 	{
-		h.append( inPlug()->fullAttributesHash( path ) );
+		for( const auto &i : gafferAttributes->members() )
+		{
+			if( shadingEngine->needsAttribute( i.first ) )
+			{
+				if ( !i.second )
+				{
+					throw Exception( "Cannot compute hash from a CompoundObject will NULL data pointers!" );
+				}
+				// \todo - why are we hashing the value instead of the address of the InternedString?
+				// More importantly, why does the default hash of CompoundObject do so much work?
+				h.append( i.first.value() );
+				i.second->hash( h );
```

Comment:
The comment about `CompoundObject::hash()` is really out of context here, where we're not even calling it - I don't think it will make much sense in future. I'd have to do a bit of a review to see whether or not we hash values instead of addresses in Gaffer - there are arguments both ways (with the argument for stable hashes being easier debugging).